### PR TITLE
SONARPY-1423 Update sonar-plugin-api to latest version (10.1.0.809)

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -183,10 +184,17 @@
...
+<!--        <version>${sonar.orchestrator.version}</version>-->
+<!--        <scope>test</scope>-->
+<!--      </dependency>-->
+
       <dependency>
         <groupId>org.sonarsource.orchestrator</groupId>
-        <artifactId>sonar-orchestrator</artifactId>
-        <version>${sonar.orchestrator.version}</version>
+        <artifactId>sonar-orchestrator-junit4</artifactId>
+        <version>4.0.0.404</version>
```
Comment (joke1196) [inline pom.xml:197]: So this is not linked to the sonar orchestrator version anymore?

File: python-frontend/src/test/java/org/sonar/plugins/python/api/PythonVersionUtilsTest.java [hunk 2]
```diff
@@ -34,7 +34,7 @@
 public class PythonVersionUtilsTest {
 
   @org.junit.Rule
-  public LogTester logTester = new LogTester();
+  public LogTester logTester = new LogTester().setLevel(Level.DEBUG);
```
Comment (joke1196) [inline python-frontend/src/test/java/org/sonar/plugins/python/api/PythonVersionUtilsTest.java:37]: I would guess here Level.WARN is enough right? No need to change it, just making sure I understand the change or if there is another reason to keep it a DEBUG level.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java [hunk 3]
```diff
@@ -343,19 +340,19 @@ private void saveMetricOnFile(InputFile inputFile, Metric<Integer> metric, Integ
...
-  private static void addQuickFixes(InputFile inputFile, RuleKey ruleKey, Iterable<PythonQuickFix> quickFixes, NewSonarLintIssue sonarLintIssue) {
+  private static void addQuickFixes(InputFile inputFile, RuleKey ruleKey, Iterable<PythonQuickFix> quickFixes, NewIssue sonarLintIssue) {
     try {
       for (PythonQuickFix quickFix : quickFixes) {
-        NewQuickFix newQuickFix = sonarLintIssue.newQuickFix()
+        var newQuickFix = sonarLintIssue.newQuickFix()
           .message(quickFix.getDescription());
 
-        NewInputFileEdit edit = newQuickFix.newInputFileEdit().on(inputFile);
+        var edit = newQuickFix.newInputFileEdit().on(inputFile);
```
Comment (joke1196) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java:355]: I am not sure I understand the reason for the type change here. Is it related to the migration?

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java [hunk 4]
```diff
@@ -267,7 +268,8 @@ public void test_execute_on_sonarlint_quickfix() throws IOException {
 
     Issue issue = context.allIssues().iterator().next();
 
-    List<SensorQuickFix> quickFixes = ((MockSonarLintIssue) issue).quickFixes;
+    var quickFixes = ((MockSonarLintIssue) issue).quickFixes();
+//    var quickFixes = ((MockSonarLintIssue) issue).quickFixes();
```
Comment (joke1196) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java:272]: I guess we can remove this comment

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java [hunk 5]
```diff
@@ -210,7 +209,9 @@ public boolean scanWithoutParsing(PythonInputFileContext inputFile) {
   private final AnalysisWarningsWrapper analysisWarning = mock(AnalysisWarningsWrapper.class);
 
   @org.junit.Rule
-  public LogTester logTester = new LogTester();
+  public LogTester logTester = new LogTester().setLevel(Level.DEBUG);
+  @org.junit.Rule
+  public LogTester traceLogTester = new LogTester().setLevel(Level.TRACE);
```
Comment (joke1196) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java:214]: Do we need both LogTester? Is it possible to make it work with the `logTester` set to TRACE level?

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java [hunk 6]
```diff
@@ -303,10 +305,10 @@ public NewQuickFix newQuickFix() {
     assertThat(issues).hasSize(1);
     MockSonarLintIssue issue = (MockSonarLintIssue) issues.iterator().next();
 
-    assertThat(issue.quickFixes).isEmpty();
-    assertThat(issue.getSaved()).isTrue();
+    assertThat(issue.quickFixes()).isEmpty();
+//    assertThat(issue.getSaved()).isTrue();
```
Comment (joke1196) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java:309]: Is this not testable anymore?

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java [hunk 7]
```diff
@@ -19,43 +19,45 @@
...
   private boolean saved;
 
   MockSonarLintIssue(SensorContextTester context) {
     this.context = context;
   }
 
   @Override
-  public NewQuickFix newQuickFix() {
-    return parent.newQuickFix();
+  public NewIssue addQuickFix(org.sonar.api.batch.sensor.issue.fix.NewQuickFix newQuickFix) {
```
Comment (joke1196) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java:54]: Maybe we could use the type name only here `NewQuickFix`

**Review by joke1196 — CHANGES_REQUESTED** [review]
Looks good! I left a few comments mainly to better understand what is happening. I was a bit confused with the change of the logger mixed with the API change for NewIssue. So I guess my main concern would be about this assertion line commented out.

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java (hunk 5 above)
Comment (maksim-grebeniuk-sonarsource) [reply sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java:214]: it could be but then it requires assertion changes

File: pom.xml (hunk 1 above)
Comment (maksim-grebeniuk-sonarsource) [reply pom.xml:197]: good catch

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java (hunk 6 above)
Comment (maksim-grebeniuk-sonarsource) [reply sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonSensorTest.java:309]: turned back

**Review by joke1196 — APPROVED** [review]
LGTM! Thanks a lot for this!

File: pom.xml [hunk 8]
```diff
@@ -90,11 +90,12 @@
...
     <mockito.version>4.11.0</mockito.version>
-    <sonar.version>9.7.1.62043</sonar.version>
-    <sonar.orchestrator.version>3.40.0.183</sonar.orchestrator.version>
-    <sonar.api.version>9.11.0.290</sonar.api.version>
-    <sonar-analyzer-commons.version>2.5.0.1358</sonar-analyzer-commons.version>
-    <sonarlint-core.version>8.14.0.63103</sonarlint-core.version>
+    <sonar.version>9.9.1.69595</sonar.version>
+    <sonar.orchestrator.version>4.0.0.404</sonar.orchestrator.version>
+    <sonar.api.version>10.1.0.809</sonar.api.version>
+    <sonar.plugin.api.version>10.0.0.695</sonar.plugin.api.version>
```
Comment (andreaguarino) [inline pom.xml:96]: why do we need two properties? I replaced usages of `sonar.plugin.api.version` with `sonar.api.version` and everything is still green

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java [hunk 9]
```diff
@@ -343,19 +340,19 @@ private void saveMetricOnFile(InputFile inputFile, Metric<Integer> metric, Integ
   }
 
   private void handleQuickFixes(InputFile inputFile, RuleKey ruleKey, NewIssue newIssue, PreciseIssue preciseIssue) {
-    if (isInSonarLint(context) && newIssue instanceof NewSonarLintIssue) {
+    if (isInSonarLint(context)) {
```
Comment (andreaguarino) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java:343]: did you check everything works fine in SonarLint? I would suggest deploying the jar locally in VSCode and test quickfixes

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/IPynbRuleRepositoryTest.java [hunk 10]
```diff
@@ -43,7 +43,7 @@ public void createRulesTest() throws IOException {
 
     assertThat(repository).isNotNull();
     assertThat(repository.language()).isEqualTo("ipynb");
-    assertThat(repository.name()).isEqualTo("SonarQube");
+    assertThat(repository.name()).isEqualTo("Sonar");
```
Comment (andreaguarino) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/IPynbRuleRepositoryTest.java:46]: I'm a bit surprised by the change of repository name. I would suggest to check if everything is still working on SonarLint analyzing a Jupyter Notebook file

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java [hunk 11]
```diff
@@ -98,8 +105,8 @@ public NewIssue addFlow(Iterable<NewIssueLocation> flowLocations) {
   }
 
   @Override
-  public NewIssue addFlow(Iterable<NewIssueLocation> flowLocations, FlowType flowType, @Nullable String flowDescription) {
-    throw new IllegalStateException("Not supposed to be tested");
+  public NewIssue addFlow(Iterable<NewIssueLocation> iterable, FlowType flowType, @Nullable String s) {
+    return null;
```
Comment (andreaguarino) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java:109]: is this change intentional? Why do we return null instead of throwing an exception?

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java [hunk 12]
```diff
@@ -151,11 +158,33 @@ public boolean isQuickFixAvailable() {
...
   }
 
-  protected boolean getSaved(){
-    return saved;
+  @Override
+  public List<QuickFix> quickFixes() {
+    return parent.quickFixes();
   }
 
+  @org.jetbrains.annotations.Nullable
```
Comment (andreaguarino) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java:169]: Usually we use `javax` annotations. For return types we use `@CheckForNull`

**Review by andreaguarino — COMMENTED** [review]
Overall looks good, Please check my comments before merging

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java (hunk 9 above)
Comment (maksim-grebeniuk-sonarsource) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonScanner.java:343]: checked, looks good

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/IPynbRuleRepositoryTest.java (hunk 10 above)
Comment (maksim-grebeniuk-sonarsource) [reply sonar-python-plugin/src/test/java/org/sonar/plugins/python/IPynbRuleRepositoryTest.java:46]: checked, looks good

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java (hunk 11 above)
Comment (maksim-grebeniuk-sonarsource) [reply sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java:109]: I just took the mock class implementation from sonar-java project.

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java (hunk 12 above)
Comment (maksim-grebeniuk-sonarsource) [reply sonar-python-plugin/src/test/java/org/sonar/plugins/python/MockSonarLintIssue.java:169]: oops, wrong import

**Review by andreaguarino — APPROVED** [review]
LGTM!
